tests/lib/rules/no-invalid-names.js

```diff
@@ -0,0 +1,232 @@
+                output: null,
+                errors: [{
+                    messageId: `${objType}NameInvalidType`,
+                    data: {
+                        name: "null",
+                        type: "object"
+                    }
+                }]
+            },
+            {
+                code: `${callee}(/regex/, function () {});`,
+                output: null,
+                errors: [{
+                    messageId: `${objType}NameInvalidType`,
+                    data: {
+                        type: "object",
+                        name: "/regex/"
+                    }
+                }]
+            },
+            {
+                code: `${callee}([], function () {});`,
+                output: null,
+                errors: [{
+                    messageId: `${objType}NameInvalidType`,
+                    data: {
+                        type: "ArrayExpression",
+                        name: "[]"
+                    }
+                }]
+            },
+            {
+                code: `${callee}({}, function () {});`,
+                output: null,
+                errors: [{
+                    messageId: `${objType}NameInvalidType`,
+                    data: {
+                        type: "ObjectExpression",
+                        name: "{}"
+                    }
+                }]
+            },
+            {
+                code: `${callee}(this, function () {});`,
+                output: null,
+                errors: [{
+                    messageId: `${objType}NameInvalidType`,
+                    data: {
+                        type: "ThisExpression",
+                        name: "this"
+                    }
+                }]
+            },
+            {
+                code: `${callee}(typeof foo, function () {});`,
+                output: null,
+                errors: [{
+                    messageId: `${objType}NameInvalidType`,
+                    data: {
+                        type: "UnaryExpression",
+                        name: "typeof foo"
+                    }
+                }]
+            },
+            {
+                code: `${callee}(void foo, function () {});`,
+                output: null,
+                errors: [{
+                    messageId: `${objType}NameInvalidType`,
+                    data: {
+                        type: "UnaryExpression",
+                        name: "void foo"
+                    }
+                }]
+            },
+            {
+                code: `${callee}(++foo, function () {});`,
+                output: null,
+                errors: [{
+                    messageId: `${objType}NameInvalidType`,
+                    data: {
+                        type: "UpdateExpression",
+                        name: "++foo"
+                    }
+                }]
+            },
+            {
+                code: `${callee}(foo + bar, function () {});`,
```

Comment:
`foo + bar` can be a legitimate way to concatenate strings so I believe it should be a valid test case.

Reply:
I suppose. I'll allow it for plus only